web/modules/custom/search_api_field/src/Plugin/Field/FieldType/SearchItem.php

```diff
@@ -0,0 +1,388 @@
+<?php
+
+namespace Drupal\search_api_field\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\search_api\Entity\Index as SearchApiIndex;
+
+/**
+ * Plugin implementation of the 'link' field type.
+ *
+ * @FieldType(
+ *   id = "search",
+ *   label = @Translation("Search"),
+ *   description = @Translation("Stores the search settings related to this field."),
+ *   default_widget = "search_default",
+ *   default_formatter = "search",
+ * )
+ */
+class SearchItem extends FieldItemBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultStorageSettings() {
+    return array(
+      'index' => NULL,
+      'facet_regions' => [],
+      'view_modes' => [],
+    ) + parent::defaultStorageSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    $properties['value'] = DataDefinition::create('any')
+      ->setLabel(t('Data'));
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function mainPropertyName() {
+    return 'value';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    return array(
+      'columns' => array(
+        'value' => array(
+          'type' => 'blob',
+          'size' => 'big',
+          'serialize' => TRUE,
+        ),
+      ),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fieldSettingsForm(array $form, FormStateInterface $form_state) {
+    return [];
+  }
```

Comment:
The parent method `FieldItemBase::fieldSettingsForm()` already returns an empty array, so you don't need to override it.